### PR TITLE
[NBS 1.0] Remove `backend-common` package

### LINKED ISSUE #26283: Attempt to entirely delete backend-common from the repo

### PR DESCRIPTION
**Pull request by freben**
Fixes #26283

This fully removes the deprecated `@backstage/backend-common` from the repo. Remaining references are by published version instead.

### CODE REVIEW

**Review by Rugvip — APPROVED** [review]
🧹 🎉

**Review by camilaibs — APPROVED** [review]
🤩 🧹

**Review by jhaals — APPROVED** [review]
-12K!!

🚀

File: packages/backend-app-api/package.json [hunk 1]
```diff
@@ -50,7 +50,7 @@
     "test": "backstage-cli package test"
   },
   "dependencies": {
-    "@backstage/backend-common": "workspace:^",
+    "@backstage/backend-common": "^0.24.1",
```
Comment (camilaibs) [inline packages/backend-app-api/package.json:53]: @freben should it be `[0.25.0-next.0](https://www.npmjs.com/package/@backstage/backend-common/v/0.25.0-next.0)` instead?

Comment (freben) [reply packages/backend-app-api/package.json:53]: Excellent callout - there were actually some interesting things in the latest next line release. Let's hold off on this until next mainline. Marked as draft and added the merge-after-release label

Comment (freben) [reply packages/backend-app-api/package.json]: These are now all updated to `^0.25.0` after the release

**Review by Rugvip — APPROVED** [review]
shipit-squirrel

**Comment by fearphage** [comment]
@freben FYI the [backend migration documentation](https://backstage.io/docs/backend-system/building-backends/migrating/) still recommends including the library that was already removed.

**Comment by freben** [comment]
Yep, for legacy backend compatibility wrappers - which live in the legacy backend package intentionally indeed.

**Pull request merged**